app/jinja_filters.py

```diff
@@ -43,6 +43,18 @@ def format_date(value):
     return value.strftime('%-d %B %Y')
 
 
+@blueprint.app_template_filter()
+def format_conditional_date(dates):
+    """ This function format_cond_date accepts two dates, if a date is submitted that is used and is available,
+    if not the metadata date is made available The format (example) in the json is
+    {{[answers.date_from, exercise.start_date]|format_cond_date}}"""
+
+    if dates[0]:
+        return datetime.strptime(dates[0], "%d/%m/%Y").strftime('%d %B %Y')
+    else:
+        return datetime.strptime(dates[1], "%d/%m/%Y").strftime('%d %B %Y')
```

Comment:
You should be able to replace these lines with calls to `format_str_as_short_date` e.g. `format_str_as_short_date(dates[0])`